universal-application-tool-0.0.1/app/controllers/PostgresController.java

```diff
@@ -35,15 +35,23 @@ public PostgresController(
             httpExecutionContext.current());
   }
 
-  public CompletionStage<Result> add(String name) {
+  public CompletionStage<Result> create(String name) {
     Person p = new Person();
     p.name = name;
     return personRepository
         .insert(p)
         .thenApplyAsync(
             id -> {
-              return ok("person " + name + " with ID: " + id.toString() + " added.");
+              return ok("person " + name + " with ID: " + id.toString() + " created.");
             },
             httpExecutionContext.current());
   }
+
+  public Result createSync(String name) {
+    Person p = new Person();
+    p.id = System.currentTimeMillis(); // not ideal, but it works
```

Comment:
The ID is autogenerated by the database and should only be set by the database.

```suggestion
```

Reply:
I can do this. However, when I tested with the change I saw ` Error[ERROR: duplicate key value violates unique constraint "pk_person"   Detail: Key (id)=(1) already exists.]` the first few times until the id incremented pass the number of sample data. Maybe I got the evolution script wrong. Let me see if I can get this right.

Reply:
Okay, by not inserting sample data with specified ids in evolution script, the issue goes away.

Reply:
Nice!